### PR TITLE
healthz parameters for cisco

### PR DESCRIPTION
**Pull request by karthikeya-remilla**
a. added the process "ifmgr" for test purposes and chassis component name "Rack 0" for Cisco
    The test intent is to kill a process and perform Healthz operations post that. So functionally any software process is valid for this purpose on Cisco IOS-XR platform
b. Origin as "openconfig" for healthz RPCs

c. refactor comps.FindStandbyRP() to comps.FindStandbyControllerCard() and it's corresponding usage in tests

### CODE REVIEW

**Comment by karthikeya-remilla** [comment]
@dplore could you please merge this if accepted?

**Review by dplore — APPROVED** [review]

**Comment by dplore** [comment]
@karthikeya-remilla this test is not passing on Cisco in our environment.  Before we can merge, I'd like to see this passing.  What are we missing?

<details><summary>Details</summary>
<p>
# show version

Wed Aug 14 15:09:11.997 PDT
Cisco IOS XR Software, Version 24.3.1.28I LNT
Copyright (c) 2013-2024 by Cisco Systems, Inc.

Build Information:
 Built By     : sajshah
 Built On     : Thu Aug 08 10:33:18 UTC 2024
 Build Host   : iox-ucs-002
 Workspace    : /auto/iox-ucs-002-san2/prod/24.3.1.28I.SIT_IMAGE/8000/ws
 Version      : 24.3.1.28I
 Label        : 24.3.1.28I-EFT2_rev2


=== RUN   TestCopyingDebugFiles
    third_party/openconfig/featureprofiles/feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go:67: 
        *** Fetching gNOI clients for xx27.den85...
           third_party/openconfig/featureprofiles/feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go:73: process bgp not found on device
--- FAIL: TestCopyingDebugFiles (18.11s)
=== RUN   TestChassisComponentArtifacts
    third_party/openconfig/featureprofiles/feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go:120: 
        *** Fetching gNOI clients for xx27.den85...
        
        
    third_party/openconfig/featureprofiles/feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go:137: Executing Healthz Check RPC for component Rack 0
    third_party/openconfig/featureprofiles/feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go:140: Unexpected error on executing Healthz Check RPC: rpc error: code = Unimplemented desc = error on request {
        path: {
          origin: "openconfig"
          elem: {
            name: "components"
          }
          elem: {
            name: "component"
            key: {
              key: "name"
              value: "Rack 0"
            }
          }
        }
        }: unknown service gnoi.healthz.Healthz
--- FAIL: TestChassisComponentArtifacts (0.76s)
FAIL


</p>
</details>

**Comment by karthikeya-remilla** [comment]
> @karthikeya-remilla this test is not passing on Cisco in our environment. Before we can merge, I'd like to see this passing. What are we missing?

This is a 24.4.1 feature. This failure is expected on 24.3.1

**Comment by karthikeya-remilla** [comment]
> Please note test is not passing in our environment and static check fix

Please review the logic while I work on static check and merge conflicts in metadata.pb.go.

**Comment by karthikeya-remilla** [comment]
updated PR description

**Comment by dplore** [comment]
Hi @karthikeya-remilla can you resolve the merge conflict with proto/metadata_go_proto/metadata.pb.go?  I will and merge this after resolved.

**Comment by karthikeya-remilla** [comment]
> Hi @karthikeya-remilla can you resolve the merge conflict with proto/metadata_go_proto/metadata.pb.go? I will and merge this after resolved.

Hi @dplore , resolved the conflicts. Could you please check now as the FR is delivered?

**Comment by dplore** [comment]
@karthikeya-remilla there are a few static check errors to be resolved.  

I am not sure why we seeing this panic in go test?   Are you seeing that when you run locally @karthikeya-remilla ?

Ref to broken github check: https://github.com/openconfig/featureprofiles/actions/runs/10744182024/job/29800536390?pr=3326#step:6:443

File: feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go
Comment (dplore) [inline feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go:93]: Please resolve static check finding:

this value of componentName is never used (SA4006)

File: internal/deviations/deviations.go
Comment (dplore) [inline internal/deviations/deviations.go:1139]: Fix static check:

exported function ChassisGetRpcUnsupported should have comment or be unexported

File: internal/deviations/deviations.go
Comment (dplore) [inline internal/deviations/deviations.go:1140]: Fix static check:

func ChassisGetRpcUnsupported should be ChassisGetRPCUnsupported

**Review by dplore — CHANGES_REQUESTED** [review]
Just a few static checks to fix, otherwise LGTM

File: feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go
Comment (dplore) [inline feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go:101]: In OC, there is only CONTROLLER_CARD. RP is a cisco specific name.

```suggestion
			standbyRpName, activeRpName := comps.FindStandbyCONTROLLER_CARD(t, dut, rpList)
```

File: feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go
Comment (dplore) [inline feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go:97]:
```suggestion
		t.Logf("Found CONTROLLER_CARD list: %v", ccList)
```

File: feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go
Comment (dplore) [inline feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go:99]: add switch statement for dut.Vendor like:
```
		switch dut.Vendor() {
		case ondatra.CISCO:
```

**Review by dplore — CHANGES_REQUESTED** [review]
Thanks for all the changes.  Couple more smaller comments, this is almost ready.

File: feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go
Comment (karthikeya-remilla) [reply feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go:99]: could you explain why the switch is needed? what would the default case be here?

the code under deviation is applicable only to Cisco at the moment. it is implied that active controller_card would be 0/RP0/CPU0 for Cisco via this block.

Comment (dplore) [reply feature/gnoi/system/tests/copying_debug_files_test/copying_debug_files_test.go:99]: It's needed because your implementation of this deviation is cisco specific.  ie: that particular deviation contains ios-xr, cisco 8800 specific strings in it.    The default case is to do nothing.  Other vendors could assert the deviation and add  cases as needed.

**Pull request merged**
